In [3]:
import pandas as pd

In [4]:
dataset=pd.read_csv("general_data.csv")
dataset.columns

Index(['Age', 'Attrition', 'BusinessTravel', 'Department', 'DistanceFromHome',
       'Education', 'EducationField', 'EmployeeCount', 'EmployeeID', 'Gender',
       'JobLevel', 'JobRole', 'MaritalStatus', 'MonthlyIncome',
       'NumCompaniesWorked', 'Over18', 'PercentSalaryHike', 'StandardHours',
       'StockOptionLevel', 'TotalWorkingYears', 'TrainingTimesLastYear',
       'YearsAtCompany', 'YearsSinceLastPromotion', 'YearsWithCurrManager'],
      dtype='object')

In [8]:
from sklearn import preprocessing
Label_encoder=preprocessing.LabelBinarizer()
dataset["Attrition"]=Label_encoder.fit_transform(dataset["Attrition"])

In [11]:
from scipy.stats import pearsonr
stats,p=pearsonr(dataset.Attrition,dataset.DistanceFromHome)
print(stats,p)

-0.009730141010179674 0.5182860428050771


__Hypothesis 1__

H0(Null Hypothesis) = There is no correlation in between Attrition & Distance From Home

Ha(Alternative Hypothesis) = There is correlation in between Attrition & Distance From Home

p value = 0.51820.5182860428050771 which is greater than 0.05

__Accept H0__

Conclusion = There is no correlation in between Attrition & Distance From Home

In [13]:
stats,p=pearsonr(dataset.Attrition,dataset.Age)
print(stats,p)

-0.15920500686577965 1.996801615886744e-26


__Hypothesis 2__

H0(Null Hypothesis) = There is no correlation in between Attrition & Age

Ha(Alternative Hypothesis) = There is correlation in between Attrition & Age

p value = 1.996801615886744e-26 which is less than 0.05

__Reject H0__

Conclusion = There is correlation in between Attrition & Distance From Home

In [14]:
stats,p=pearsonr(dataset.Attrition,dataset.Education)
print(stats,p)

-0.015111167710968713 0.3157293177118575


__Hypothesis 3__

H0(Null Hypothesis) = There is no correlation in between Attrition & Education

Ha(Alternative Hypothesis) = There is correlation in between Attrition & Education

p value = 0.3157293177118575 which is greater than 0.05

__Accept H0__

Conclusion = There is no correlation in between Attrition & Education

In [19]:
stats,p=pearsonr(dataset.Attrition,dataset.PercentSalaryHike)
print(stats,p)

0.03253259489105349 0.030743386433355353


__Hypothesis 4__

H0(Null Hypothesis) = There is no correlation in between Attrition & PercentSalaryHike

Ha(Alternative Hypothesis) = There is correlation in between Attrition & PercentSalaryHike

p value = 0.030743386433355353 which is less than 0.05

__Reject H0__

Conclusion = There is correlation in between Attrition & PercentSalaryHike

In [20]:
stats,p=pearsonr(dataset.Attrition,dataset.YearsAtCompany)
print(stats,p)

-0.1343922139899772 3.1638831224877484e-19


__Hypothesis 5__

H0(Null Hypothesis) = There is no correlation in between Attrition & YearsAtCompany

Ha(Alternative Hypothesis) = There is correlation in between Attrition & YearsAtCompany

p value = 3.1638831224877484e-19 which is less than 0.05

__Reject H0__

Conclusion = There is correlation in between Attrition & YearsAtCompany

In [22]:
stats,p=pearsonr(dataset.Attrition,dataset.YearsSinceLastPromotion)
print(stats,p)

-0.03301877514258434 0.028330336189396753


__Hypothesis 6__

H0(Null Hypothesis) = There is no correlation in between Attrition & YearsSinceLastPromotion

Ha(Alternative Hypothesis) = There is correlation in between Attrition & YearsSinceLastPromotion

p value = 0.028330336189396753 which is less than 0.05

__Reject H0__

Conclusion = There is correlation in between Attrition & YearsSinceLastPromotion

In [24]:
stats,p=pearsonr(dataset.Attrition,dataset.YearsWithCurrManager)
print(stats,p)

-0.15619931590162847 1.7339322652896276e-25


__Hypothesis 7__

H0(Null Hypothesis) = There is no correlation in between Attrition & YearsWithCurrManager

Ha(Alternative Hypothesis) = There is correlation in between Attrition & YearsWithCurrManager

p value = 1.7339322652896276e-25 which is less than 0.05

__Reject H0__

Conclusion = There is correlation in between Attrition & YearsWithCurrManager

In [25]:
stats,p=pearsonr(dataset.Attrition,dataset.MonthlyIncome)
print(stats,p)

-0.031176281698115007 0.03842748490600132


__Hypothesis 8__

H0(Null Hypothesis) = There is no correlation in between Attrition & MonthlyIncome

Ha(Alternative Hypothesis) = There is correlation in between Attrition & MonthlyIncome

p value = 0.03842748490600132 which is less than 0.05

__Reject H0__

Conclusion = There is correlation in between Attrition & MonthlyIncome